torchmultimodal/models/coca/multimodal_decoder.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Callable, Optional
+
+import torch
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.transformer import TransformerDecoder
+from torchmultimodal.utils.attention import get_causal_attention_mask
+
+
+class CoCaMultimodalDecoder(nn.Module):
+    """
+    Multimodal decoder with cross-attention for CoCa model.
+    Based on the implementation in open_clip: https://tinyurl.com/mn35vdmd
```

Comment:
It would be useful to have a brief description of the architecture and use cases for this model here. Is the CoCA decoder a variant on a more general architecture?

Reply:
It is pretty much just a generic transformer decoder with cross-attention and optional output projection. I can add more detail here though